test/unittest/layers/layers_standalone_common_tests.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file layer_common_tests.cpp
+ * @date 15 June 2021
+ * @brief Common test for nntrainer layers (Param Tests)
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <layers_common_tests.h>
+
+#include <app_context.h>
+#include <layer_devel.h>
+
+constexpr unsigned SAMPLE_TRIES = 10;
+
+TEST_P(LayerSemantics, createFromAppContext_pn) {
+  auto &ac = nntrainer::AppContext::Global();
+  EXPECT_EQ(ac.createObject<nntrainer::Layer>(expected_type)->getType(),
+            expected_type);
+}
+
+TEST_P(LayerSemantics, setProperties_p) {
+  /// @todo check if setProperties does not collide with layerNode designated
+  /// properties
+}
+
+TEST_P(LayerSemantics, setProperties_n) {
+  /** must not crash */
+  EXPECT_THROW(layer->setProperty({"unknown_props=2"}), std::invalid_argument);
+}
+
+TEST_P(LayerSemantics, setPropertiesValidWithInvalid_n) {}
+
+TEST_P(LayerSemantics, setPropertiesValidInvalidOnly_n) {}
+
+TEST_P(LayerSemantics, finalizeOutputValidate_p) {
+  nntrainer::TensorDim in_dim({1, 1, 1, 1});
+  nntrainer::InitLayerContext init_context =
+    nntrainer::InitLayerContext({in_dim});
+  init_context.validate();
+
+  // set necessary properties only
+  EXPECT_NO_THROW(layer->setProperty(valid_properties));
+
+  if (!must_fail) {
```

Comment:
I understand that some rare layers are meant to fail, but in this case, I think it's better for the layer to have it's own `test` instead of using `test_p`, unless obeying the semantics.

Reply:
I will leave this for now, as creating another set of tests for this layer will be cumbersome and reduce maintainability.
As this layer can be created and added by the user, the full testsuite has to be run.
I will find another way to handle this later.

Reply:
Okay